python/tvm/ansor/workload_registry.py

```diff
@@ -0,0 +1,170 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Workload registration and serialization.
+
+We use a json string to represent a workload (a computation graph).
+The format of the string is `[func_name, [args...]]`.
+The dag should be the return value of this `func_name(*args)`.
+
```

Comment:
I am trying to understand this...Could you check if the following understanding is correct?

A. "Workload" is the workload ansor wants to tune. For exmaple, it can be a matmul, whose inputs are n, m, k (matrix dimensions).
B. The workload key is used to describe the entire workload, i.e. the entire workload can be completely described using registered function and its inputs. For example, using `n`, `m`, `k`, we can reconstruct the matmul function by re-creating the two corresponding tensors, build a ComputeOp and then return its result.
C. The computational dag can be created using a workload key.

Reply:
That's right! The workload key will be as part of the tuning record, so that we can easily replay the results from log.